### PR TITLE
Fix wrong annotation while creating drcluster MW

### PR DESCRIPTION
**Pull request by ELENAGER**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2139835

### CODE REVIEW

**Review by ShyamsundarR — COMMENTED** [review]
Add checks for annotation values (and maybe any labels) [here](https://github.com/RamenDR/ramen/blob/9bd126d5a446ad8c6911df8e43678dd0a46ca11d/controllers/drcluster_controller_test.go#L143-L170)

We should have caught this in integration tests, so let's get that going as well.

File: controllers/drclusters.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func drClusterDeploy(drClusterInstance *drclusterInstance, ramenConfig *rmn.Rame
 
 	annotations := make(map[string]string)
 
-	annotations["DRClusterName"] = mwu.InstName
+	annotations[DRClusterNameAnnotation] = mwu.InstName
```
Comment (nirs) [inline controllers/drclusters.go:51]: Is this leftover before we added `DRClusterNameAnnotation`?

Comment (ELENAGER) [reply controllers/drclusters.go:51]: No, this is from the same [PR](https://github.com/RamenDR/ramen/pull/478/)

**Comment by ELENAGER** [comment]
@ShyamsundarR I added annotation verification [here](https://github.com/RamenDR/ramen/pull/1236/commits/b87ddc5fb3f83c66605b9880b4afaeab3b5105b2)

File: controllers/drclusters.go (hunk 1 above)
Comment (nirs) [reply controllers/drclusters.go:51]: OK, I guess it slipped in the review, this was a very big change.

File: controllers/drcluster_controller_test.go [hunk 2]
```diff
@@ -154,18 +154,24 @@ func validateClusterManifest(drcluster *ramen.DRCluster, disabled bool) {
 		expectedCount = 4
 	}
 
+	clusterName := drcluster.Name
+
+	key := types.NamespacedName{
+		Name:      util.DrClusterManifestWorkName,
+		Namespace: clusterName,
+	}
```
Comment (nirs) [inline controllers/drcluster_controller_test.go:162]: Much nicer like this!

File: controllers/drcluster_controller_test.go [hunk 3]
```diff
@@ -154,18 +154,24 @@ func validateClusterManifest(drcluster *ramen.DRCluster, disabled bool) {
 		expectedCount = 4
 	}
 
+	clusterName := drcluster.Name
```
Comment (nirs) [inline controllers/drcluster_controller_test.go:157]: Do we really need this? Why not use drcluster.Name directly?

**Review by ShyamsundarR — APPROVED** [review]

**Review by BenamarMk — APPROVED** [review]
Good catch

**Pull request merged**